### DISCUSSION TITLE
ESLint couldn't determine the plugin "import" uniquely.

### DISCUSSION THREAD

**Discussion by warnellw** [vercel/next.js#28775] [Help] [answered]
Hey all,

I'm running into an issue with my project's ESLint configuration after installing the Next.js ESLint plugin.

My project is a mono-repo, and all apps extend a base ESLint configuration file. The `import` plugin is already used and configured in the mono-repo's base ESLint config. After installing Next.js's ESLint plugin, and adding it to the Next.js app's `.eslintrc.json` (which extends the base ESLint config), I get the following error:

`ESLint couldn't determine the plugin "import" uniquely.`

Does anyone know how I can resolve this conflict? Unfortunately it's not possible to remove the `import` plugin from the mono-repo's base `.eslintrc.json` because it's used by every other app and lib in the mono-repo.

Thank you!

Reproduction:

Root config: `.eslintrc.json`
```
{
  "root": true,
  "plugins": ["import"],
  "extends": ["plugin:import/recommended"],
  "rules": {
    // some more import rules here
  }
  ...
}
```

Next.js app level config: `apps/nextjs/.eslintrc.json`
```
{
  "extends": [
    "../../.eslintrc.json",
    "next",
    "next/core-web-vitals"
  ]
}
```

**Comment by JustFly1984** [comment, marked as answer]
@warnellw I had similar issues with other eslint plugins. The issue is that some eslint configs, or other dependencies are using other versions to your package.json
I had only one solution: use yarn, and add `resolutions` field to package.json with specific version to override version to the one you need.

**Reply by warnellw** [reply]
Thank you! There is a version mismatch: `eslint-config-next` is using v2.24.2 of `eslint-plugin-import`, while my mono-repo is on v2.22.1.

**Reply by warnellw** [reply]
Following up here: updating my mono-repo to `eslint-plugin-import` v2.24.2 (matching the version used by `eslint-config-next`) and regenerating the `yarn.lock` file has fixed the issue. I will need to look into `resolutions` because my mono-repo uses pinned package versions, and `eslint-plugin-next` is using a ranged version. I assume this will break again sometime in the near future.
